docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+    * Sum numbers in a vector.
+    *
+    * @param values Container whose values are summed.
+    * @return sum of `values`, or 0.0 if `values` is empty.
+    */
+   double sum(std::vector<double> & const values) {
+       ...
+   }
+
+Not:
+
+.. code-block:: cpp
+
+   double sum(std::vector<double> & const values) {
+       /**
+        * Sum numbers in a vector.
+        *
+        * @param values Container whose values are summed.
+        * @return sum of `values`, or 0.0 if `values` is empty.
+        */
+       ...
+   }
+
+.. _cpp-doxygen-sections:
+
+Common Structure of Documentation Blocks
+========================================
+
+We organize Doxygen comment blocks into sections that appear in a common order. This format is analogous to the one adopted for the :ref:`Python documentation <py-docstring-sections>`.
+The sections and their relative order are:
+
+1. :ref:`cpp-doxygen-short-summary`
+2. :ref:`cpp-doxygen-deprecation` (if applicable)
+3. :ref:`cpp-doxygen-extended-summary` (recommended)
+4. :ref:`cpp-doxygen-tparameters` (if applicable; for classes, methods, and functions)
+5. :ref:`cpp-doxygen-parameters` (if applicable; for methods and functions)
+6. :ref:`cpp-doxygen-returns` (if applicable; for methods and functions)
+7. :ref:`cpp-doxygen-throws` (if applicable; for methods and functions)
+8. :ref:`cpp-doxygen-exceptsafe` (for methods and functions)
+9. :ref:`cpp-doxygen-related` (if applicable; for functions)
+10. :ref:`cpp-doxygen-initializer` (if applicable; for constants)
+11. :ref:`cpp-doxygen-see-also` (optional)
+12. :ref:`cpp-doxygen-notes` (optional)
+13. :ref:`cpp-doxygen-references` (optional)
+14. :ref:`cpp-doxygen-examples` (optional)
+
+For summaries of how these sections are composed in specific contexts, see:
+
+- :ref:`cpp-doxygen-package-definition`
+- :ref:`cpp-doxygen-class-structure`
+- :ref:`cpp-doxygen-enum-structure`
+- :ref:`cpp-doxygen-method-function-structure`
+- :ref:`cpp-doxygen-attribute-constants-structure`
+
+.. _cpp-doxygen-short-summary:
+
+Short Summary
+-------------
+
+A one-line summary that does not use variable names or the function's name. This summary will appear in lists of class/namespace members.
+
+.. code-block:: cpp
+
+   /**
+    * Sum two numbers.
+    */
+   double add(double a, double b);
+
+For functions and methods, the summary should be written in the imperative voice (i.e., as a command that the API consumer is giving). Getters and other methods that are more naturally described as values rather than actions may ignore this rule.
+
+.. _cpp-doxygen-deprecation:
+
+Deprecation Warning
+-------------------
+
+A ``@deprecated`` tag (where applicable) to warn users that the component is deprecated. The text that follows the tag should include:
+
+1. In what stack version the object was deprecated, and when it will be removed.
+2. Reason for deprecation if this is useful information (e.g., object is superseded, duplicates functionality found elsewhere, etc.).
```

Comment:
With the `e.g.,` you can leave out the `etc.`

(I need fix this in https://developer.lsst.io/docs/py_docs.html#deprecation-warning 😬!)